Minio/DataModel/VersioningArgs.cs

```diff
@@ -0,0 +1,125 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using Minio.DataModel;
+using Minio.Exceptions;
+using RestSharp;
+
+namespace Minio
+{
+    public class GetVersioningInfoArgs : BucketArgs
+    {
+        public GetVersioningInfoArgs()
+        {
+        }
+        public new GetVersioningInfoArgs WithBucket(string bucket)
+        {
+            this.BucketName = bucket;
+            return this;
+        }
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+
+        public RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            RestRequest request = RequestUtil.CreateRequest(baseUrl, Method.GET, authenticator, this.BucketName, this.Secure, this.Region);
+            request.AddQueryParameter("versioning","");
+            return request;
+        }
+
+        public VersioningConfiguration ProcessResponse(RestSharp.IRestResponse response)
+        {
+            VersioningConfiguration config = null;
+            if (HttpStatusCode.OK.Equals(response.StatusCode))
+            {
+                using (var stream = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(response.Content)))
+                {
+                    config = (VersioningConfiguration)new XmlSerializer(typeof(VersioningConfiguration)).Deserialize(stream);
+                }
+                return config;
+            }
+            return null;
+        }
+    }
+
+    public class SetVersioningArgs : BucketArgs
+    {
+        public new SetVersioningArgs WithBucket(string bucket)
+        {
+            this.BucketName = bucket;
+            return this;
+        }
+
+        public SetVersioningArgs()
+        {
+        }
+
+
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+            if ( this.VersioningSuspended && !this.VersioningEnabled )
+            {
+                throw new InvalidBucketOperationException(this.BucketName, "Versioning has to be enabled to be suspended.");
```

Comment:
this doesn't make sense. VersioningEnabled and VersioningSuspended are toggled states - you don't need 2 variables to represent that.

Reply:
Implemented based on a previous conversation with @kannappanr . States are Versioning not Enabled (versioning configuration is null, hence both flags false), Versioning Enabled and not suspended (enabled flag is true, suspended is false), Versioning Enabled but suspended (enabled is true, suspended is also true). So, suspended true & enabled false is an illegitimate status. Please do let me know if this implementation is correct.

Reply:
can you rename this.VersioningEnabled -> this.Versioned to make it clearer?

Reply:
Change done.